implementation/src/main/java/io/smallrye/config/ConfigMappingProvider.java

```diff
@@ -67,14 +68,16 @@ final class ConfigMappingProvider implements Serializable {
         this.validateUnknown = builder.validateUnknown;
 
         final ArrayDeque<String> currentPath = new ArrayDeque<>();
+        final NameIterator nameIterator = NameIterator.empty();
         for (Map.Entry<String, List<Class<?>>> entry : roots.entrySet()) {
```

Comment:
@franz1981 IIRC, this kind of iteration is not great. Would using `Maps.forEach` be better?

Reply:
I didn't focus on the collections here, tbh, but will take a look! Usually forEach is the best but when the code is hot, and not cold (at interpreted/C1) compilation level... lambda metafactory is terribly slow if the compilation Is not C2!
Indeed for interpreted/C1 compilation usually worth avoid lambdas and operations, in general: the less is better (which goes a bit against what I have said at https://github.com/smallrye/smallrye-config/pull/984#issuecomment-1694765001 sigh!)

Reply:
How about if MethodHandle was being used?

Reply:
Mmmm same thing likely, these methods usually have some not negligible bootstrap cost...which is the reason why in cold cases, I prefer Atomic updaters vs VarHandle (which requires method handles)

Reply:
It's not easy but if you are curious (I am), I can distill some microbench on it, wdyt?

Reply:
Not if you have better things to do :)

Reply:
We always have, but this one could be a good direction to improve out start-up times; it was sometime I wanted to write some "do/don't" guide to help Devs to write good optimized code for startup, which sadly need to have slightly different style in the "core" parts we care

Reply:
👌🏼